jetbrains-ultimate/tst/software/aws/toolkits/jetbrains/utils/rules/GoCodeInsightTestFixtureRule.kt

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.utils.rules
+
+import com.goide.GoConstants
+import com.goide.sdk.GoSdkType
+import com.intellij.openapi.projectRoots.Sdk
+import com.intellij.openapi.projectRoots.impl.ProjectJdkImpl
+import com.intellij.testFramework.LightProjectDescriptor
+import com.intellij.testFramework.PsiTestUtil
+import com.intellij.testFramework.fixtures.CodeInsightTestFixture
+import com.intellij.testFramework.fixtures.IdeaTestFixtureFactory
+
+class GoCodeInsightTestFixtureRule : CodeInsightTestFixtureRule() {
+    override fun createTestFixture(): CodeInsightTestFixture {
+        val fixtureFactory = IdeaTestFixtureFactory.getFixtureFactory()
+        val projectFixture = fixtureFactory.createLightFixtureBuilder(GoLightProjectDescriptor())
```

Comment:
Pass GoLightProjectDescriptor to the CodeInsightTestFixtureRule constructor to avoid leaking virtual file pointers and then call super to make the fixture

Reply:
That's in pr 3 on top of this :P

Reply:
Oh boy.im still on pt1..